### ISSUE TITLE
Quantile Regression

### ISSUE THREAD

**Issue by rjbergerud** [benjaminmielke/pod-challenge#11]
One of the top 4 papers https://arxiv.org/pdf/1809.03561.pdf in Gefcom 2017 used this approach.  It's relatively simple compared to some other methods, so would be worth trying on this dataset.

Calculating quantiles as opposed to a single predicted value would be useful for both charging and discharging.  For instance, hourly irradiance predictions for partially cloudy could have delayed or early predictions of cloud cover.  What this means is that for some periods of sun the quantiles will differ drastically $$(q_{.25}, q_{.75} = (0, max_sun)$$ while for others, they'll be fairly tight bounds.  We can use this to prioritize charging during times only when the lower quantile $q_{.25}$ is high if we know the sum of these lower quantiles over the day is greater than our 6MWh battery.

**Comment by benjaminmielke** [comment]
This is very interesting, and we should definitely pursue it to give it a go. I found a python package developed by, i tihnk Microsoft, called LightGBM that specializes in quantile regression forecasting. 

check an article about it here: http://ethen8181.github.io/machine-learning/ab_tests/quantile_regression/quantile_regression.html

**Comment by rjbergerud** [comment]
Ok, I'll make it my plan to dig into this on the weekend

**Comment by benjaminmielke** [comment]
There is also this post that outlines procedures to complete Quantile Regression using TensorFlow(standard neural network) or LightGBM(which is tree based gradient boost). I think trying both the tree based and neural network for the QR would be a solid start. I can look into the TensorFlow solution

https://medium.com/the-artificial-impostor/quantile-regression-part-2-6fdbc26b2629

**Comment by rjbergerud** [comment]
Sure, we can take one of each, I'll have a go at LightGBM then.

**Comment by rjbergerud** [comment]
Just adding a note to clarify my thinking on this. 
Our current strategy for charging is approximately/akin to take a proportion of the .5 quantile so that battery reaches full charge by 3pm.
A better strategy for charging than our current strategy might be to find the quantile for which the area under the curve is 6MWh, and charge at the rate given by that quantile.  

The motivation is comes from imagining a situation where we get a large number of megawatts quantile q_{\alpha} where \alpha is small (say less than .1).  We'd want to charge at least the rate of this quantile, because it's a pretty sure bet we'll get this amount, and we can't be rewarded by getting more than 6MWh, which means we can't make a higher risk "bet" for a big payback in MWh.

If we get evenly-spaced quantiles predictions, we could make a mock probability distribution function, and then solve the constrained expectation-maximization problem of 
E(solar_charge) = \int p(x) \max(\min(x,y),2.5)
where y is the charge rate, x power output the panel.  Once the expected value is above 6MWh, we want to solve for the charge rate which has the least left-side variance.
